skimage/morphology/binary.py

```diff
@@ -28,9 +47,13 @@ def binary_erosion(image, footprint=None, out=None):
     ----------
     image : ndarray
         Binary input image.
-    footprint : ndarray, optional
+    footprint : ndarray or tuple, optional
         The neighborhood expressed as a 2-D array of 1's and 0's.
-        If None, use a cross-shaped footprint (connectivity=1).
+        If None, use a cross-shaped footprint (connectivity=1). This can also
+        be a sequence of 2-tuples where the first element of each 2-tuple is a
+        footprint and the second element as an integer describing the number of
+        times it should be iterated.
+        in ``skimage.morphology.footprints``.
```

Comment:
IMHO this reads a little bit weird. Line 56 is loose here.

This:
> This can also be a sequence of 2-tuples where the first element of each 2-tuple is a footprint and the second element as an integer describing the number of times it should be iterated.

Could be in the `Notes`, like:
> `footprint` can also be a sequence (...)

And an example of that. What do you think?

Reply:
@alexdesiqueira, What do you think of the following? 

First in the parameters section, a fairly brief description

```
    footprint : ndarray or tuple, optional
        The neighborhood expressed as a 2-D array of 1's and 0's.
        If None, use a cross-shaped footprint (connectivity=1). The footprint
        can also be provided as a sequence of smaller footprints as described
        in the notes below.
```

and then in the notes:
```
    Notes
    -----
    The footprint can also be a provided as a sequence of 2-tuples where the
    first element of each 2-tuple is a footprint ndarray and the second element
    is an integer describing the number of times it should be iterated. For
    example specifying ``footprint=[(np.ones((5, 1)), 1), np.ones((1, 5), 1)]``
    would apply a 5x1 footprint followed by application of a 1x5 footprint
    giving a net effect that is the same as ``footprint=np.ones((5, 5))``, but
    with lower computational cost. Many of the builtin footprints such as
    ``skimage.morphology.disk`` provide an option to automically generate a
    footprint sequence of this type.
```

Reply:
@grlee77 looks great, thanks!

Reply:
Okay I have made this change to the binary and grayscale morphology functions and fixed all the conflicts with the prior PR that was merged